server/events/project_command_runner.go

```diff
@@ -429,6 +449,11 @@ func (p *DefaultProjectCommandRunner) doApply(ctx models.ProjectCommandContext)
 	}
 	defer unlockFn()
 
+	// Drop request if a more recent VCS event updated Atlantis state
+	if p.StaleCommandChecker.CommandIsStale(ctx, p.PullStatusFetcher) {
```

Comment:
Why are we doing this in ProjectCommandRunner? That means we'd be doing this check (db read) for every project we find. Can we do this one level above? In the command runner itself?

Reply:
Moved to command_runner.go